src/diffpy/morph/morphs/morphfuncy.py

```diff
@@ -0,0 +1,65 @@
+from diffpy.morph.morphs.morph import LABEL_GR, LABEL_RA, Morph
+
+
+class MorphFuncy(Morph):
+    """Apply the user-supplied Python function to the y-coordinates of the
+    morph data"""
+
+    # Define input output types
+    summary = "Apply a Python function to the y-axis data"
+    xinlabel = LABEL_RA
+    yinlabel = LABEL_GR
+    xoutlabel = LABEL_RA
+    youtlabel = LABEL_GR
+
+    def morph(self, x_morph, y_morph, x_target, y_target):
+        """General morph function that applies a user-supplied function to the
+        y-coordinates of morph data to make it align with a target.
+
+        Configuration Variables
+        -----------------------
+        function: callable
+            The user-supplied function that applies a transformation to the
+            y-coordinates of the data.
+
+        parameters: dict
+            A dictionary of parameters to pass to the function.
+            These parameters are unpacked using **kwargs.
+
+        Returns
+        -------
+            A tuple (x_morph_out, y_morph_out, x_target_out, y_target_out)
+            where the target values remain the same and the morph data is
+            transformed according to the user-specified function and parameters
+            The morphed data is returned on the same grid as the unmorphed data
+
+        Example
+        -------
+        Import the funcy morph function:
+        >>> from diffpy.morph.morphs.morphfuncy import MorphFuncy
+        Define or import the user-supplied transformation function:
```

Comment:
I wonder if this would be easier to read with judiciously placed blank lines, maybe after each code block, but play around till it looks nice?  I am not sure if there is a docstring standard for this, in which case follow it, but if not, then I think a few blank lines could increase readability.